server/controllers/comment.controller.js

```diff
@@ -34,13 +35,27 @@ function create(req, res, next) {
  * @property {number} req.query.limit - Limit number of users to be returned.
  * @returns {[Comment]}
  */
-function list(req, res, next) {
-  const { postId } = req.params;
-  Comment.getCommentsForItem(postId)
-    .then((comments) => {
-      res.json({result: comments});
-    })
-    .catch(e => next(e));
-}
+ function list(req, res, next) {
+   const { postId } = req.params;
+
+   Comment.getTopLevelCommentsForItem(postId)
+   .then((comments) => {
+     // Here we are fetching our nested comments, and need everything to finish
+     let nestedCommentPromises = _.map(comments, (comment) => {
+       return Comment.fillNestedComments(comment);
+     });
+     return Promise.all(nestedCommentPromises);
```

Comment:
This will be refactored to aggregate all ids of top level comments and fetch their children in one query instead of comments.length queries.